### PR TITLE
[#165145794] Implement Authentication Modal

### PR DESCRIPTION
**Pull request by MandelaK**
**Description**
<hr />
This PR adds an authentication modal to the project. The Modal can switch between login form and register form through an action dispatch and it can be called from anywhere through a dispatch
<br/><br/>

**Type of Change**
<hr />

- [x] feature(non-breaking change).
<br/>

**Checklist**
<hr />

- [x] Setup modal redux state management
- [x] Create the modal
<br/>

**Related Stories**
<hr />

[165145794](https://www.pivotaltracker.com/n/projects/2313280/stories/165145794)

### CODE REVIEW

File: src/components/Header.jsx
Comment (MandelaK) [inline src/components/Header.jsx]: Because we don't need to pass any payload, we can just pass the type only instead of passing a payload of null.

File: src/components/Header.jsx
Comment (MandelaK) [inline src/components/Header.jsx]: No need of passing a payload of null, just pass the type only and exclude payload altogether, ie `dispatch({ type: REGISTER })`

File: src/store/reducers/index.js
Comment (MandelaK) [inline src/store/reducers/index.js]: Please fix the typo `paylod`

**Comment by MandelaK** [comment]
Good job, the modal seems to be working, please fix the minor errors I have pointed out, let's rally the team to review this so we can finish the feature

**Comment by cob04** [comment]
I have tested it manually. I think it works great.

**Review by BrianSerem — DISMISSED** [review]
Unless you add more changes this is good to go.

File: src/components/AuthenticationModal.jsx
Comment (abulojoshua1) [inline src/components/AuthenticationModal.jsx]: @MandelaK, kindly destructure state here.

File: src/components/AuthenticationModal.jsx
Comment (abulojoshua1) [inline src/components/AuthenticationModal.jsx]: @MandelaK, use `const` instead of `let` when you will not change the value of a variable.

File: src/components/Header.jsx
Comment (abulojoshua1) [inline src/components/Header.jsx]: @MandelaK, kindly add space after `}` on this line.

File: src/css/App.css
Comment (abulojoshua1) [inline src/css/App.css]: @MandelaK, remove this commented out code if it is not necessary.

File: src/css/App.css
Comment (abulojoshua1) [inline src/css/App.css]: @MandelaK, use percentages or other values for heights instead of pixels, pixels can act unpredictably on different screen sizes.

File: src/store/actions/action-types.js
Comment (abulojoshua1) [inline src/store/actions/action-types.js]: @MandelaK, add EOF here.

File: src/tests/components/AuthenticationModal.test.js
Comment (abulojoshua1) [inline src/tests/components/AuthenticationModal.test.js]: @MandelaK, make the `it` block descriptive, use something like `it (' should blah blah blah..', () => {
})`

File: src/tests/components/AuthenticationModal.test.js
Comment (abulojoshua1) [inline src/tests/components/AuthenticationModal.test.js:42]: @MandelaK, also make sure the number of tests you write provides considerable test coverage for the code you have written.

**Review by abulojoshua1 — CHANGES_REQUESTED** [review]

**Comment by abulojoshua1** [comment]
@MandelaK, add more info on what the PR does.

File: src/components/AuthenticationModal.jsx
Comment (MandelaK) [inline src/components/AuthenticationModal.jsx]: remember to remove this `console.log` statement before it is merged

File: src/store/actions/change-form-action.js
Comment (MandelaK) [inline src/store/actions/change-form-action.js]: Export this as `default`

**Review by RachelleMaina — APPROVED** [review]
Apart from the issues that had been pointed out and hound violations, which have been solved already, everything looks good to me

**Comment by MandelaK** [comment]
Everything looks good to me ... Apparently since I am the one who opened this PR, I cannot approve it.

**Review by Etomovich — APPROVED** [review]

**Review by ElMonstro — APPROVED** [review]

**Review by cob04 — APPROVED** [review]
I have tested it manually and I find it satisfactory. I also like the API you have introduced it makes it easy to integrate social login.

File: src/components/AuthenticationModal.jsx
Comment (ElMonstro) [reply src/components/AuthenticationModal.jsx:13]: I have used prop-types I don't know why it's displaying this

**Review by ElMonstro — APPROVED** [review]

File: src/middleware/auth-modal-middleware.js
Comment (abulojoshua1) [inline src/middleware/auth-modal-middleware.js]: This file contains actions, kindly change place it in the actions directory and rename it to something line AuthActions.js

**Review by abulojoshua1 — CHANGES_REQUESTED** [review]

**Comment by abulojoshua1** [comment]
@ElMonstro, use camel case for js file naming. For example `camelCase.js`

**Review by abulojoshua1 — APPROVED** [review]
LGTM

**Pull request merged**
